backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -238,16 +278,133 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
     }
   }
 
-  /* Blacklist by prefix if appropriate. */
-  private def handleForbidden[T](path: String, andThen: => State): State = {
+  /* Blacklist by bucket and hit if appropriate. */
+  private def handleBlacklistingForForbidden[T](path: String, andThen: => State): State = {
     for {
+      // Blacklist the hit first in the forcomp since not all configurations will support bucket blacklisting.
       cache <- standardParams.blacklistCache
+      data <- stateData
+      _ = blacklistAndMetricHit(cache, data.cacheHit)
       prefix <- extractBlacklistPrefix(path)
-      _ = cache.blacklist(prefix)
+      _ = blacklistAndMetricBucket(cache, prefix)
     } yield()
     andThen
   }
 
+  private def handleBlacklistingForGenericFailure(): Unit = {
```

Comment:
[TOL] Could we make lines 294-407 a helper class that gets a reference to the cache? Would make the actor more focused on actor-ey state transitions.

Reply:
ooh yes...I did a mechanical refactor to extract to a trait and that helped a ton. 🙂